lane.py

```diff
@@ -1393,34 +1398,256 @@ def search_target(self):
                       languages=languages, media=media, audiences=audiences)
         return wl
 
-    def groups(self, _db):
-        """Extract a list of samples from each child of this Lane, as well as
-        from the lane itself. This can be used to create a grouped
-        acquisition feed for the Lane.
+    def featured_window(self, target_size):
+        """Randomly select an interval over `Work.random` that ought to
+        contain approximately `target_size` high-quality works from
+        this lane.
+
+        :param: A 2-tuple (low value, high value), or None if the
+        entire span should be considered.
+        """
+        if self.size < 100:
+            # This lane is so small that there's a significant risk
+            # that the interval might not contain enough works. Just
+            # choose randomly from the entire lane.
+            return 0,1
+        width = target_size / (self.size * 0.2)
+        
+        maximum_offset = 1-width
+        start = min(random.random(), maximum_offset)
+        return start, start+width
+
+    def groups(self, _db, include_sublanes=True):
+        """Return a list of (Lane, MaterializedWork) 2-tuples
+        describing a sequence of featured items for this lane and
+        (optionally) its children.
```

Comment:
What's the use case for optional children? I was thinking this generated the featured books for the children and then the 'All X' lane, but I would think 'All X' is the optional part of that.

Reply:
When you're building a groups feed for a lane, and one of its sublanes needs to have its featured feed calculated separately, you call sublane.groups(_db, include_sublanes=False). You don't want the sublanes because you're not building a groups feed for the sublane, you're building the sublane's _portion_ of its parent's feed.

Reply:
Oh I see, that's for the lanes that don't inherit parent restrictions.